### ISSUE TITLE
Can be used to transform not Javax to Jakarta?

### ISSUE THREAD

**Issue by Thihup** [eclipse/transformer#116]
I'd like to use the transformer to migrate from Jackson 1 to Jackson 2, but I tried to use the CLI and I couldn't figure it out.
I don't know if the CLI supports some input file with the packages to change or it is only the packages that are builtin.

Thanks

**Comment by jgallimore** [comment]
Hi @Thihup ! I need to have another pass at this, which I'll try and do today. I did contribute the full set of packages to transform a while back, so hopefully what you need is in the project.

Are you looking transform Jackson itself, or your project that is using Jackson?

It sounds like your use-case is to transform your project, and to swap out one version of Jackson for another. Let me know if I have that right or not.

**Comment by Thihup** [comment]
>It sounds like your use-case is to transform your project, and to swap out one version of Jackson for another. Let me know if I have that right or not.

You're right! My current project is using the Jackson 1 and I would like to try the Transformer to see if the application still would work as expected using the Jackson 2.

Thanks!

**Comment by jgallimore** [comment]
Can you shoot over the Maven co-ordinates for the two versions of Jackson you're using?

**Comment by Thihup** [comment]
I'm using the Jackson 1.9.6 and would like to try the latest Jackson (2.11.3).

https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-core-asl/1.9.6
https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-mapper-asl/1.9.6
https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-xc/1.9.6
https://mvnrepository.com/artifact/org.codehaus.jackson/jackson-jaxrs/1.9.6

**Comment by tbitonti** [comment]
Hi, is this still an active question?

**Comment by Thihup** [comment]
> Hi, is this still an active question?

Yes. I want to transform all uses of the old package name from Jackson 1 to Jackson 2. But I couldn't figure it out how to do it. It seems that it only supports migrating from javax to jakarta.

**Comment by tbitonti** [comment]
Hi, can you make a list of the initial and final package names?
You should be able to do this by providing a local "package renames" properties file.
Here is an example of one which is used by open-liberty:
https://github.com/OpenLiberty/open-liberty/blob/integration/dev/wlp-jakartaee-transform/rules/jakarta-renames.properties
For example, with package renames property file "jackson.renames.properties",
when invoking the transformer, add this to the invocation:
-tr jackson.renames.properties
Note that you will want to run with the basic version of the transformer, that is, which does not have the default jakarta rules.  Otherwise, you will also be performing jakarta updates.  Unless you want to do that too.  That is, use main class "org.eclipse.transformer.Transformer".  If you use the main class "org.eclipse.transformer.jakarta.JakartaTransformer" that will include the default jakarta rules data.

And, there is a bug in the rules data handling, such that at least one text update file must be provided.

Here is a minimal set of properties files:

```
properties\renames.properties
--currently empty
-- update this with package rename values
properties\text.properties
-- update this with:
"ignoreMe=dummy.properties"
dummy.properties
--currently empty
```

Then invoke the transformer with:

`java -cp org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar org.eclipse.transformer.Transformer input output -v -tr properties\renames.properties -tf properties\text.properties`

"-v" is useful initially, and can be removed to trim down the output.

The invocation assumes that the dependeny jars are in a local "libs" folder, and that "properties" is also a local folder.
